cmake/TBB.cmake

```diff
@@ -25,7 +25,17 @@ include("cmake/Threading.cmake")
 
 macro(handle_tbb_target)
     if(TBB_FOUND)
-        set_property(TARGET TBB::tbb PROPERTY "MAP_IMPORTED_CONFIG_RELWITHMDD" "DEBUG")
+        if(WIN32)
+            # On Windows we must link to debug version of TBB library to ensure ABI compatibility
+            # with MSVC debug runtime.
+            set_property(TARGET TBB::tbb PROPERTY "MAP_IMPORTED_CONFIG_RELWITHMDD" "DEBUG")
+        else()
```

Comment:
wondering if this should be limited to SYCL only maybe?
and also, does it mean that for older TBB (the ones w/a config files) we still continue link to DEBUG?